provider/tencentcloud/cloudapi/mockapi.go

```diff
@@ -71,7 +71,7 @@ func (api *mockAPIService) DeletePrivateZoneRecord(request *privatedns.DeletePri
 	}
 	for _, privateZoneRecord := range api.privateZoneRecords[*request.ZoneId] {
 		deleteflag := false
-		if request.RecordIdSet != nil && len(request.RecordIdSet) != 0 {
+		if len(request.RecordIdSet) != 0 {
```

Comment:
This makes sense, but is it needed for the upgrade to 1.23 or are we mixing different changes? If so, I'd prefer to split them.

Reply:
This is required with linter update. 
Linter says that len(x) is already returning 0 when x is nil.

Reply:
This is how we proceeded for go1.22 or 1.21: update go & linter with code accordingly for green ci.